src/index.js

```diff
@@ -1,37 +1,118 @@
 import './style.css';
+import { dateElement, options, today } from './modules/date.js';
+import { LIST } from './modules/depen.js';
 
-const list = document.querySelector('.form-items');
+dateElement.innerHTML = today.toLocaleDateString('en-US', options);
+let userInput = [];
 
-const myList = [
-  {
-    listItem: 'Play piano at 5',
-    completed: false,
-    id: 1,
-  },
-  {
-    listItem: 'Watch olympic race at 7',
-    completed: false,
-    id: 2,
-  },
-  {
-    listItem: 'Go roller blading with Tony',
-    completed: false,
-    id: 3,
-  },
-  {
-    listItem: 'Dance lesson at 12 noon',
+function renderTodo(todo) {
+  localStorage.setItem('todoItemsRef', JSON.stringify(userInput));
+  const list = document.querySelector('.form-items');
+  const item = document.querySelector(`[data-key='${todo.id}']`);
+
+  if (todo.deleted) {
+    item.remove();
+    if (userInput.length === 0) list.innerHTML = '';
+    return;
+  }
+
+  const isCompleted = todo.completed ? 'complete' : '';
+  const htmlLi = document.createElement('li');
+
+  htmlLi.setAttribute('class', `input-li ${isCompleted}`);
+  htmlLi.setAttribute('data-key', todo.id);
+  htmlLi.innerHTML = `
+  <input id="${todo.id}" type="checkbox"/>
+  <label class="ticked js-tick" for="${todo.id}"></label>
+      <p class="task input-li item-edit render-item" id="${todo.id}">${todo.listItem}</p>
+  <button class="delete-todo js-delete-todo"><img src="https://img.icons8.com/external-wanicon-lineal-wanicon/64/000000/external-delete-user-interface-wanicon-lineal-wanicon.png" class="delete-todo js-delete-todo alt="delete" id="delete"/></button>
+  
+  <img class="edit" src="https://img.icons8.com/external-dreamstale-lineal-dreamstale/32/000000/external-edit-interface-dreamstale-lineal-dreamstale-2.png" alt="edit" />
+  `;
+  if (item) {
+    list.replaceChild(htmlLi, item);
+  } else {
+    list.append(htmlLi);
+  }
+}
+
+function addTodo(text) {
+  const todo = {
+    listItem: text,
     completed: false,
-    id: 4,
-  },
-];
+    id: Date.now(),
+  };
+
+  userInput.push(todo);
+  renderTodo(todo);
+}
 
-let htmlList = '';
+function toggleDone(key) {
+  const index = userInput.findIndex((item) => item.id === Number(key));
+  userInput[index].completed = !userInput[index].completed;
+  renderTodo(userInput[index]);
+}
 
-myList.forEach((item) => {
-  htmlList += `<li class="input-li"><input type="checkbox" name="inputDone" id="inputDone">
-    <p class="item-edit" id="${item.id}">${item.listItem}</p>
-    <img class="grab" src="https://img.icons8.com/material-outlined/24/000000/menu-2.png" alt="grab">
-  </li>`;
+function deleteTodo(key) {
+  const index = userInput.findIndex((item) => item.id === Number(key));
+  const todo = {
+    deleted: true,
+    ...userInput[index],
+  };
+  userInput = userInput.filter((item) => item.id !== Number(key));
+  renderTodo(todo);
+}
+
```

Comment:
- [ ] Kindly note that when I am deleting one task sometimes is cleaning the arrays in your localstorage. Kindly fix this issue. check the screen video below 👇 

https://user-images.githubusercontent.com/65068771/151587526-39af9129-c95c-4afb-99eb-86b688545aec.mp4

Reply:
Thank you for your reply.
I have made the required changes. 
All test pass.

https://user-images.githubusercontent.com/81635839/151604003-238a1a93-112c-4f7a-9a21-b37d403f3007.mov